dashboard/lib/services/lesson_import_helper.rb

```diff
@@ -0,0 +1,279 @@
+    activity_section.remarks = true
+    description = match[1].strip
+    slide_matches = find_slides(description)
+    if slide_matches.empty? || description.index(slide_matches[0][0]) != 0
+      activity_section.description = description.strip
+    else
+      activity_section.description = description.delete_prefix(slide_matches[0][0])
+      activity_section.slide = true
+    end
+    activity_section
+  end
+
+  def self.find_tips(markdown)
+    # See https://github.com/code-dot-org/remark-plugins/blob/master/src/tip.js
+    # Looks for tips that look like "!!!tip <tip-0>" followed by text. It will
+    # capture the text until it hits either the end of the string or a line that
+    # starts with a non-whitespace character.
+    regex = /^!!! *?([\w-]+)(?: "(.*?)")?(?: <(.*?)>)?(?:[\s]+$)+([\d\D]+?)(?=(^\S|^$))/
+    markdown.to_enum(:scan, regex).map {Regexp.last_match}
+  end
+
+  def self.find_tip_links(markdown)
+    # See https://github.com/code-dot-org/remark-plugins/blob/master/src/tiplink.js
+    # Looks for the location where tip icons should appear
+    # Example: tip!!!tip-0<!-- place where you'd like the icon --> some text
+    # <!-- place where you'd like the icon --> is optional but is written out
+    # in this regex in order to be able to correctly the text that should be displayed
+    regex = /^([\w-]+)!!! ?([\w-]+)(?:<!-- place where you'd like the icon -->)?(.+?)$/
+    markdown.to_enum(:scan, regex).map {Regexp.last_match}
+  end
+
+  # Removes tabs (or 4 spaces) at the beginning of lines
+  def self.format_tip_markdown(markdown)
+    markdown = markdown.chomp.reverse.chomp.reverse
+    markdown.gsub(/(^\t|^ {4})/, '')
+  end
+
+  def self.create_tip(type, key, markdown)
+    tip_map = {
+      "tip" => "teachingTip",
+      "content" => "contentCorner",
+      "discussion" => "discussionGoal",
+      "assessment" => "assessmentOpportunity"
+    }
+    {key: key, type: tip_map[key], markdown: format_tip_markdown(markdown)}
+  end
+
+  def self.create_activity_section_with_tip(tip_link_match, tip_match_map)
+    tip = tip_match_map[tip_link_match[2]].shift
+    unless tip
+      return ActivitySection.new(description: tip_link_match[3].strip)
+    end
+    tip_match = tip[:match]
+    tip_match_map.delete(tip_link_match[2]) if tip_match
+    activity_section = ActivitySection.new
+
+    # Sometimes theres the the slide icon here as well. Check for that and apply if needed.
+    description = tip_link_match[3].strip
+    slide_matches = find_slides(description)
+    if slide_matches.empty? || description.index(slide_matches[0][0]) != 0
+      activity_section.description = description.strip
+    else
+      activity_section.description = description.delete_prefix(slide_matches[0][0])
+      activity_section.slide = true
+    end
+
+    activity_section.tips = [create_tip(tip_match[3] || "#{tip_match[1]}-0", tip_match[1] || "tip", tip_match[4])]
+    activity_section
+  end
+
+  # No levels, no tips, just markdown
```

Comment:
so this means right now if we find a header we are making it its own activity section and not combining it in an activity section with some of the stuff below it?

Reply:
Yeah currently :/ Now that I'm thinking through this more, this might actually be a relatively straightforward fix though, so I'll try to fix this today in a separate PR.